src/main/java/dino/command/Command.java

```diff
@@ -0,0 +1,15 @@
+package dino.command;
+
+import dino.exception.*;
+import dino.task.TaskList;
+import dino.util.Storage;
+
+/**
+ * Contains a set of instructions that can be understood and executed by the ChatBot
+ */
+public abstract class Command {
+
+    abstract public String execute(Storage storage, TaskList taskList) throws InvalidFormatException, TimeNotSpecifiedException, EmptyTaskDescriptionException, InvalidIndexException, TaskAlreadyDoneException, InvalidInputException, IndexNotSpecifiedException, EmptyListException, TaskNotFoundException;
```

Comment:
Perhaps you can have a parent class like DinoException, where all these exceptions extend from this DinoException class. Then the method signature could be throws DinoException instead, of having such a long method signature. I think it would be quite a headache, when you add more exceptions and will have to change the method signatures of your command classes.

Reply:
After reading further down, I realize you do have a DinoException, so why not just throws DinoException instead? :)